keyv/Map.h

```diff
@@ -0,0 +1,359 @@
+     * enabled by setting a non-zero queue depth. Applications then need to
+     * quarantee that the inserted values stay valid until 'depth' other
+     * elements have been inserted or flush() has been called. Implementations
+     * which do not support asynchronous writes return 0.
+     *
+     * @return the queue depth chosen by the implementation, smaller or equal to
+     *         the given depth.
+     * @version 1.11
+     */
+    KEYV_API size_t setQueueDepth( const size_t depth );
+
+    /**
+     * Insert or update a value in the database.
+     *
+     * @param key the key to store the value.
+     * @param value the value stored at the key.
+     * @return true on success, false otherwise
+     * @throw std::runtime_error if the value is not copyable
+     * @version 1.9.2
+     */
+    template< class V > bool insert( const std::string& key, const V& value )
+        { return _insert( key, value, std::is_trivially_copyable< V >( ));}
+    KEYV_API bool insert( const std::string& key, const void* data,
+                              size_t size );
+
+    /**
+     * Insert or update a vector of values in the database.
+     *
+     * @param key the key to store the value.
+     * @param values the values stored at the key.
+     * @return true on success, false otherwise
+     * @throw std::runtime_error if the vector values are not copyable
+     * @version 1.9.2
+     */
+    template< class V >
+    bool insert( const std::string& key, const std::vector< V >& values )
+        { return _insert( key, values, std::is_trivially_copyable< V >( ));}
+
+    /**
+     * Insert or update a set of values in the database.
+     *
+     * @param key the key to store the value.
+     * @param values the values stored at the key.
+     * @return true on success, false otherwise
+     * @throw std::runtime_error if the set values are not copyable
+     * @version 1.9.2
+     */
+    template< class V >
+    bool insert( const std::string& key, const std::set< V >& values )
+        { return insert( key, std::vector<V>( values.begin(), values.end( ))); }
+
+    /**
+     * Retrieve a value for a key.
+     *
+     * @param key the key to retrieve.
+     * @return the value, or an empty string if the key is not available.
+     * @version 1.9.2
+     */
+    KEYV_API std::string operator [] ( const std::string& key ) const;
+
+    /**
+     * Retrieve a value for a key.
+     *
+     * @param key the key to retrieve.
+     * @return the value, or an empty string if the key is not available.
+     * @version 1.11
+     */
+    template< class V > V get( const std::string& key ) const
+        { return _get< V >( key ); }
+
+    /**
+     * Retrieve a value as a vector for a key.
+     *
+     * @param key the key to retrieve.
+     * @return the values, or an empty vector if the key is not available.
+     * @version 1.9.2
+     */
+    template< class V > std::vector< V > getVector( const std::string& key )
```

Comment:
Here I agree. 

In general it's a style question: I prefer one line when it fits, but will yield to the majority.